CMakeLists.txt

```diff
@@ -0,0 +1,934 @@
+	endforeach()
+add_custom_target(po-gen ALL DEPENDS ${po_gen})
+
+endif()
+
+
+#to help with the install
+list(TRANSFORM git_shell_scripts PREPEND "${CMAKE_BINARY_DIR}/")
+list(TRANSFORM git_perl_scripts PREPEND "${CMAKE_BINARY_DIR}/")
+
+#install
+install(TARGETS git git-shell
+		RUNTIME DESTINATION bin)
+install(PROGRAMS ${CMAKE_BINARY_DIR}/git-cvsserver  #${CMAKE_SOURCE_DIR}/gitk-git/gitk check
+		DESTINATION bin)
+
+set(bin_symlinks
+git-receive-pack git-upload-archive git-upload-pack)
+
+
+foreach(b ${bin_symlinks})
+install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink git${EXE} ${CMAKE_INSTALL_PREFIX}/bin/${b}${EXE})")
+endforeach()
+
+#On windows to create relative symbolic links you have to do this
+if(WIN32)
+	install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ..\\\\..\\\\bin\\\\git${EXE} ${CMAKE_INSTALL_PREFIX}/libexec/git-core/git${EXE})")
+	install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ..\\\\..\\\\bin\\\\git-shell${EXE} ${CMAKE_INSTALL_PREFIX}/libexec/git-core/git-shell${EXE})")
+else()
+	install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ../../bin/git${EXE} ${CMAKE_INSTALL_PREFIX}/libexec/git-core/git${EXE})")
+	install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ../../bin/git-shell${EXE} ${CMAKE_INSTALL_PREFIX}/libexec/git-core/git-shell${EXE})")
+endif()
+
+foreach(b ${git_symlinks})
+	string(REPLACE "${CMAKE_BINARY_DIR}" "" b ${b})
+	if(WIN32)
+		install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ..\\\\..\\\\bin\\\\git${EXE} ${CMAKE_INSTALL_PREFIX}/libexec/git-core/${b})")
+	else()
+		install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink ../../bin/git${EXE} ${CMAKE_INSTALL_PREFIX}/libexec/git-core/${b})")
+	endif()
+endforeach()
+
+foreach(b ${git_http_symlinks})
+	string(REPLACE "${CMAKE_BINARY_DIR}" "" b ${b})
+	install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink git-remote-http${EXE} ${CMAKE_INSTALL_PREFIX}/libexec/git-core/${b})")
+endforeach()
+
+list(REMOVE_ITEM PROGRAMS_BUILT git git-shell)
+install(TARGETS ${PROGRAMS_BUILT}
+		RUNTIME DESTINATION libexec/git-core)
+
+install(PROGRAMS ${git_shell_scripts} ${git_perl_scripts} ${CMAKE_BINARY_DIR}/git-p4
+		DESTINATION libexec/git-core)
+
+install(DIRECTORY mergetools DESTINATION libexec/git-core)
+install(DIRECTORY ${CMAKE_BINARY_DIR}/perl/build/lib/ DESTINATION share/perl5
+	FILES_MATCHING PATTERN "*.pm")
+install(DIRECTORY ${CMAKE_BINARY_DIR}/templates/blt/ DESTINATION share/git-core/templates)
+
+if(MSGFMT_EXE)
+	install(DIRECTORY ${CMAKE_BINARY_DIR}/locale DESTINATION share)
+endif()
+
+
+
+if(BUILD_TESTING)
+#TODO exclude tests which are not supported
+message(WARNING 
+		"To run tests make the following changes\n"
+		"OPTION 1\n"
+		"Change line 37 t/test-lib.sh -> set GIT_BUILD_DIR to your build directory\n"
+		"OPTION 2\n"
+		"Do the build in the source directory")	
```

Comment:
Alternatively, we could modify `t/test-lib.sh` to accept e.g. a pre-set `GIT_BUILD_DIR`. In a preparatory commit, for example.

Reply:
will do it.